astropy/io/fits/util.py

```diff
@@ -671,8 +671,12 @@ def _array_to_file_like(arr, fileobj):
 
     if hasattr(np, 'nditer'):
         # nditer version for non-contiguous arrays
-        for item in np.nditer(arr):
-            fileobj.write(item.tostring())
+        if arr.flags.f_contiguous:
+            for item in np.nditer(arr.T, order='F'):
+                fileobj.write(item.tostring())
+        else:
+            for item in np.nditer(arr):
+                fileobj.write(item.tostring())
```

Comment:
So here I think that we need to force C order in all cases.

Reply:
indeed, that's a lot more elegant and it works !